src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/ItemsFrozenSet.cs

```diff
@@ -22,11 +23,19 @@ protected ItemsFrozenSet(HashSet<T> source, bool optimizeForReading = true) : ba
 
             _items = new T[entries.Length];
 
+            int[] arrayPoolHashCodes = ArrayPool<int>.Shared.Rent(entries.Length);
+            Span<int> hashCodes = arrayPoolHashCodes.AsSpan(0, entries.Length);
+            for (int i = 0; i < entries.Length; i++)
+            {
+                hashCodes[i] = entries[i] is T t ? Comparer.GetHashCode(t) : 0;
+            }
+
             _hashTable = FrozenHashTable.Create(
-                entries.Length,
-                index => entries[index] is T t ? Comparer.GetHashCode(t) : 0,
+                hashCodes,
                 (destIndex, srcIndex) => _items[destIndex] = entries[srcIndex],
```

Comment:
Not entirely familiar with the implementation, but curious why we're doing delayed initialization for the `_items` array?

Reply:
It's not delayed, it's happening during the call to Create, which is helping to put the values in the right spot.